src/background/Wallet/Wallet.ts

```diff
@@ -203,6 +204,8 @@ export class Wallet {
     this.record = await this.walletStore.read(this.id, this.userCredentials);
     if (this.record) {
       this.emitter.emit('recordUpdated');
+      chainConfigStore.cleanupDeprecatedCustomChainPermissions(this);
+      chainConfigStore.checkChainsForUpdates(this);
```

Comment:
This is too much of a coupling. Why call two methods if we only ever call them together?

Reply:
We decided to not call `chainConfigStore` here at all